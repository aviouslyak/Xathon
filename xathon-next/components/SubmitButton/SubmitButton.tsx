import React from "react";

interface Props extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  className?: string;
}
const SubmitButton: React.FC<Props> = ({ className, ...props }) => {
  return (
    <button
      className={`${className} button w-40 h-10 text-xl`}
      type="submit"
      {...props}
    >
      {props.children}
    </button>
  );
};

export default SubmitButton;
