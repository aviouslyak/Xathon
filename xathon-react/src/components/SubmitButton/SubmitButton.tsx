import React from "react";

interface Props extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  className?: string;
}
const SubmitButton: React.FC<Props> = ({ className, ...props }) => {
  return (
    <button
      className={`${className} accent-gradient rounded-full w-1/5 p-2 cursor-pointer`}
      {...props}
    >
      {props.children}
    </button>
  );
};

export default SubmitButton;
