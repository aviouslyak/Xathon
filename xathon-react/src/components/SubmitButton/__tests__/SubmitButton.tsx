import { render, screen } from "@testing-library/react";
import SubmitButton from "../SubmitButton";

describe("<SubmitButton />", () => {
  beforeEach(() => {
    render(
      <SubmitButton>
        <p>foo</p>
      </SubmitButton>
    );
  });

  test("matches snapshot", () => {
    const submitComponent = screen.getByRole("button");
    expect(submitComponent).toMatchSnapshot();
  });

  test("renders children", () => {
    const pElement = screen.getByText(/foo/i);
    expect(pElement).toBeInTheDocument();
  });
});
