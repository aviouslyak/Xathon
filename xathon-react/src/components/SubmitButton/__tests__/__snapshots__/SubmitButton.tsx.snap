// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubmitButton /> matches snapshot 1`] = `
<button
  class="undefined accent-gradient rounded-full w-1/5 p-2 cursor-pointer"
  type="submit"
>
  <p>
    foo
  </p>
</button>
`;
